src/kOS/Serialization/IDumperWithSharedObjects.cs

```diff
@@ -0,0 +1,11 @@
+﻿using System;
+using kOS.Safe.Encapsulation;
+
+namespace kOS.Serialization
+{
+    public interface IDumperWithSharedObjects : IDumper
+    {
+        void SetSharedObjects(SharedObjects sharedObjects);
```

Comment:
if we are exposing currently private properties/fields of implementing classes. I would prefer turning this into a property setter requirement.